tinyagentos/scheduler/gpu_arbiter.py

```
@@ -0,0 +1,473 @@
+
+def _default_vram_probe() -> tuple[int, int]:
+    """No-GPU fallback: returns (0, 0)."""
+    return 0, 0
+
+
+def _probe_nvidia_vram() -> tuple[int, int]:
+    """Probe free/total VRAM from nvidia-smi. Returns (free_mb, total_mb)."""
+    try:
+        import subprocess
+        free_raw = subprocess.run(
+            ["nvidia-smi", "--query-gpu=memory.free", "--format=csv,noheader,nounits"],
```

Ruff (0.15.20)

[error] Starting a process with a partial executable path

(S607)